test/org/traccar/protocol/Arnavi4FrameDecoderTest.java

```diff
@@ -0,0 +1,65 @@
+package org.traccar.protocol;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.traccar.ProtocolTest;
+
+import java.nio.ByteOrder;
+
+public class Arnavi4FrameDecoderTest extends ProtocolTest {
+
+    @Test
+    public void testDecodeValidPackets() throws Exception {
+
+        Arnavi4FrameDecoder decoder = new Arnavi4FrameDecoder();
+
+        Assert.assertEquals( // Valid HEADER v1 packet with IMEI
+                binary(ByteOrder.LITTLE_ENDIAN, "ff22f30c45f5c90f0300"),
+                decoder.decode(null, null, binary(ByteOrder.LITTLE_ENDIAN, "ff22f30c45f5c90f0300")));
+
+        Assert.assertEquals( // Valid HEADER v2 packet with IMEI
+                binary(ByteOrder.LITTLE_ENDIAN, "ff23f30c45f5c90f0300"),
+                decoder.decode(null, null, binary(ByteOrder.LITTLE_ENDIAN, "ff23f30c45f5c90f0300")));
+
+        Assert.assertEquals( // Valid PACKAGE with answer to server on file transfer
```

Comment:
I just noticed this... why are valid packages decoded into "null". That should not be the case.